packages/nix/mkDevShells/pf-env-bootstrap-suspend.sh

```diff
@@ -0,0 +1,302 @@
+    echo "No Auto Scaling Groups found for EKS cluster '$EKS_CLUSTER_NAME' with nodegroup starting with '$nodegroup_prefix'."
+    return
+  fi
+
+  for asg in $asgs; do
+    echo "$asg" --min-size "$min_size" --desired-capacity "$desired_capacity"
+    echo "Updating ASG: $asg"
+
+    if aws autoscaling update-auto-scaling-group --auto-scaling-group-name "$asg" --min-size "$min_size" --desired-capacity "$desired_capacity"; then
+      echo "Successfully updated $asg to min-size $min_size and desired-capacity $desired_capacity."
+    else
+      echo "Failed to update $asg."
+    fi
+  done
+}
+
+scale_asg_controller_nodes 0 0
+
+terminate_controller_nodes() {
+  echo "Listing all EKS controller nodes..."
+
+  # Get all controller nodes by filtering on the label for EKS node groups (assuming controller nodes have specific labels)
+  # Adjust the label as per your nodegroup's labeling scheme.
+  nodes=$(kubectl "$CONTEXT_ARGS" get nodes -l panfactum.com/class=controller -o json | jq -r '.items[].metadata.name')
+
+  if [ -z "$nodes" ]; then
+    echo "No controller nodes found."
+  else
+    echo "Found controller nodes, starting termination process..."
+
+    # Loop through each node
+    for node in $nodes; do
+      # Get the providerID for the node
+      provider_id=$(kubectl "$CONTEXT_ARGS" get node "$node" -o json | jq -r '.spec.providerID')
+
+      # Extract instance_id from the providerID (format: aws:///us-east-2a/i-00f7cadf689464acd)
+      instance_id=$(echo "$provider_id" | awk -F '/' '{print $NF}')
+
+      echo "Terminating EC2 instance $instance_id"
+
+      # Terminate the EC2 instance
+      aws ec2 terminate-instances --instance-ids "$instance_id"
+
+      echo "Waiting for instance termination..."
+      aws ec2 wait instance-terminated --instance-ids "$instance_id" --no-cli-pager
+
+      echo "EC2 instance $instance_id terminated successfully."
+    done
+
+    echo "All EKS controller nodes have been terminated."
+  fi
+}
+
+terminate_controller_nodes
+
+####################################################################
+# Step 7: Scale down 'nat-' ASGs
+####################################################################
+scale_asgs_nat_nodes() {
+  local prefix="nat-"
+  local min_size=$1
+  local desired_capacity=$2
+
+  asgs=$(aws autoscaling describe-auto-scaling-groups | jq -r ".AutoScalingGroups[] | select(.AutoScalingGroupName | startswith(\"$prefix\")) | .AutoScalingGroupName")
+
+  if [ -z "$asgs" ]; then
+    echo "No Auto Scaling Groups found that start with '$prefix'."
+    return
+  fi
+
+  for asg in $asgs; do
+    echo "Updating ASG: $asg"
+
+    if aws autoscaling update-auto-scaling-group --auto-scaling-group-name "$asg" --min-size "$min_size" --desired-capacity "$desired_capacity"; then
+      echo "Successfully updated $asg to min-size $min_size and desired-capacity $desired_capacity."
+    else
+      echo "Failed to update $asg."
+    fi
+  done
+}
+
+scale_asgs_nat_nodes 0 0
```

Comment:
I feel that there should be a warning and confirmation here as well as the NAT nodes are not necessarily specific to the cluster. Deleting these will disable outbound networking in the entire VPC. I can see a scenario where a user would not want this.